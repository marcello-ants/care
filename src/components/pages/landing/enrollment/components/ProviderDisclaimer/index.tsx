// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
