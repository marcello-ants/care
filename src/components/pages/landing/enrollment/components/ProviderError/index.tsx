// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
