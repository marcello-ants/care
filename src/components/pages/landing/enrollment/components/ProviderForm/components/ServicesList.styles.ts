import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
