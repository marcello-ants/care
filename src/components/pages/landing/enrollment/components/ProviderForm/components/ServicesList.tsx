/* eslint-disable import/prefer-default-export */
// External Dependencies
import { useState } from 'react';
import { Select, MenuItem, useMediaQuery } from '@material-ui/core';
import { useFormikContext } from 'formik';
import { theme } from '@care/material-ui-theme';
import { Typography } from '@care/react-component-lib';

// Internal Constants
import { PROVIDER_SERVICES_DESKTOP, PROVIDER_SERVICES_MOBILE } from '../constants';

// Styles
import useStyles from './ServicesList.styles';

function ServicesList() {
  // Styles
  const classes = useStyles();
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'));

  const formik = useFormikContext<{ providerService: string }>();
  const listOfServices = isDesktop ? PROVIDER_SERVICES_DESKTOP : PROVIDER_SERVICES_MOBILE;

  // Component's State
  const [typeOfService, setTypeOfService] = useState('');

  // Handlers
  const onServiceValueChange = (value: string) => {
    formik.setFieldValue('providerService', value);
    setTypeOfService(value as string);
  };

  return (
    <>
      <Select
        name="selectJobType"
        className={
          !typeOfService
            ? `${classes.inputContainer} ${classes.empty}`
            : `${classes.inputContainer}`
        }
        value={typeOfService}
        displayEmpty
        onChange={(e) => {
          onServiceValueChange(e.target.value as string);
        }}>
        <MenuItem value="" disabled>
          Select one:
        </MenuItem>
        {listOfServices.map((service: any) => {
          return (
            <MenuItem value={service.value} key={service.value}>
              {service.label}
            </MenuItem>
          );
        })}
      </Select>
      {formik.errors.providerService && (
        <Typography careVariant="info1" className={classes.error}>
          Please select a job
        </Typography>
      )}
    </>
  );
}

export { ServicesList };
