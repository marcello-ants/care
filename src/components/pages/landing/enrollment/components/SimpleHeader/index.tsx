// External Dependencies
import { Link } from '@care/react-component-lib';
import Button from '@material-ui/core/Button';
import { IconCareLogo } from '@care/react-icons';

// Styles
import useStyles from './styles';

export default function SimpleHeader() {
  // Styles
  const classes = useStyles();
  const headerClasses = classes.header;

  return (
    <>
      <header className={headerClasses}>
        <div className={classes.headerContent}>
          <div>
            <IconCareLogo width="120px" height="32px" />
          </div>
          <div>
            <>
              <div className={classes.loginDesktop}>
                <Link careVariant="body3" href="https://www.care.com/app/vhp/get-started">
                  Looking for a caregiver?
                </Link>
                <Button
                  color="secondary"
                  variant="outlined"
                  size="small"
                  className={classes.loginLink}
                  href="https://www.care.com/login">
                  Member login
                </Button>
              </div>
              <div className={classes.loginMobile}>
                <Link careVariant="body3" href="https://www.care.com/login">
                  Login
                </Link>
              </div>
            </>
          </div>
        </div>
      </header>
    </>
  );
}
