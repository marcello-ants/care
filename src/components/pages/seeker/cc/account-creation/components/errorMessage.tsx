import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
