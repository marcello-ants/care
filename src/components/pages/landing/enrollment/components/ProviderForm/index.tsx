// External Dependencies
import { useCallback } from 'react';
import getConfig from 'next/config';
import { FormikProvider } from 'formik';
import Button from '@material-ui/core/Button';
import { Box, Hidden } from '@material-ui/core';
import { Typography } from '@care/react-component-lib';

// Internal Dependencies
import FormikInlineTextField from '@/components/blocks/FormikInlineTextField';
import EmailField from '@/components/features/accountCreation/EmailField';
import PasswordField from '@/components/features/accountCreation/PasswordField';
import { HearAboutUsField } from '@/components/HearAboutUs';
import { useProviderEnrollmentForm } from './utils/formHandler';
import { ServicesList } from './components/ServicesList';
import ProviderError from '../ProviderError';
import ProviderDisclaimer from '../ProviderDisclaimer';

// Styles
import useStyles from './styles';

// Constants
const {
  publicRuntimeConfig: { CZEN_GENERAL_LOGIN },
} = getConfig();

function SeekerForm() {
  // Styles
  const classes = useStyles();

  // Formik
  const { formik, isSubmitError } = useProviderEnrollmentForm();

  // Handlers
  const handleFormSubmission = useCallback(() => {
    if (!formik.isValid || formik.isSubmitting) {
      return;
    }

    formik.handleSubmit();
  }, []);

  return (
    <>
      {isSubmitError && <ProviderError />}
      <FormikProvider value={formik}>
        <Typography variant="h2" className={classes.sectionTitle}>
          <span>What job are you most</span>{' '}
          <Hidden smUp>
            <br />
          </Hidden>{' '}
          <span>interested in?</span>
        </Typography>
        <ServicesList />
        <Box mt={5}>
          <Typography variant="h2" className={`${classes.sectionTitle}`}>
            <span>Create your account</span>
            <Typography careVariant="disclaimer1" className={classes.sectionTitleDisclaimer}>
              *required
            </Typography>
          </Typography>
        </Box>
        <Box className={classes.inputContainer}>
          <EmailField
            id="email"
            name="email"
            label="Email*"
            czenGeneralLoginUrl={CZEN_GENERAL_LOGIN}
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputsContainer}>
          <PasswordField
            id="password"
            name="password"
            label="Password*"
            defaultHelperText=""
            showErrorWhenTouched
          />
        </Box>
        <Box className={classes.inputContainer} mb={2}>
          <FormikInlineTextField id="providerZip" name="providerZip" label="Zip code*" />
        </Box>
        <Box className={classes.inputContainer}>
          <HearAboutUsField name="howDidYouHearAboutUs" />
        </Box>
        <Box mb={4}>
          <ProviderDisclaimer />
        </Box>
        <Button
          color="primary"
          variant="contained"
          size="large"
          className={classes.submitButton}
          disabled={formik.isSubmitting}
          onClick={handleFormSubmission}>
          Join now
        </Button>
      </FormikProvider>
      <Typography variant="body1" className={classes.companyInquiries}>
        <span>For company inquiries</span>{' '}
        <a href="https://www.care.com/business-p1369-q59964639.html">sign up here</a>
        <span>.</span>
      </Typography>
    </>
  );
}

export default SeekerForm;
