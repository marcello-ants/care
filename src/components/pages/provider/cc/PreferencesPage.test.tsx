import React from 'react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { render, fireEvent, waitFor, act, screen } from '@testing-library/react';
import { useRouter } from 'next/router';
import { AppStateProvider } from '@/components/AppState';
import { ThemeProvider } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';
import { initialAppState } from '@/state';
import { cloneDeep } from 'lodash-es';
import { FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import Preferences from '@/components/pages/provider/cc/PreferencesPage';
import {
  GET_CAREGIVER_ENROLLMENT_STATUS,
  CAREGIVER_ATTRIBUTES_UPDATE,
} from '@/components/request/GQL';
import {
  CLIENT_FEATURE_FLAGS,
  CZEN_BACKGROUND_CHECK_CC,
  PROVIDER_CHILD_CARE_ROUTES,
} from '@/constants';

const initialStateClone = cloneDeep(initialAppState);

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

initialStateClone.flow = {
  ...initialAppState.flow,
  userHasAccount: true,
};

const featureFlagsMockTrue = {
  [CLIENT_FEATURE_FLAGS.PROVIDER_CC_FREE_GATED_EXPERIENCE]: {
    value: true,
    reason: {
      kind: 'FALLTHROUGH',
    },
  },
};

const caregiverMock = {
  request: {
    query: CAREGIVER_ATTRIBUTES_UPDATE,
    variables: {
      input: {
        childcare: {
          ageGroups: 'EARLY_SCHOOL',
          careForSickChild: true,
          numberOfChildren: 1,
        },
        serviceType: 'CHILD_CARE',
      },
    },
  },
  result: {
    data: {
      caregiverAttributesUpdate: {
        __typename: 'CaregiverAttributesUpdateSuccess',
        dummy: 'Success',
      },
    },
  },
};

const completenessIncomplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: false,
        bio: false,
        education: false,
        jobInterest: false,
        languages: false,
        overallStatus: 'INCOMPLETE',
        photo: false,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessJobInterestComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: false,
        bio: false,
        education: false,
        jobInterest: true,
        languages: false,
        overallStatus: 'INCOMPLETE',
        photo: false,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessAvailabilityComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: false,
        education: false,
        jobInterest: true,
        languages: false,
        overallStatus: 'INCOMPLETE',
        photo: false,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessProfileComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: false,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: false,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessBioComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: false,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessPhotoComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: true,
        qualities: false,
        services: false,
        subscriptionPlanViewed: false,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessFreeGatedHourlyRate = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: true,
        qualities: false,
        services: false,
        subscriptionPlanViewed: true,
        freeGated: {
          hourlyRate: false,
          appDownload: false,
          welcomeBack: false,
        },
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessFreeGatedAppDownload = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: true,
        qualities: false,
        services: false,
        subscriptionPlanViewed: true,
        freeGated: {
          hourlyRate: true,
          appDownload: false,
          welcomeBack: false,
        },
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessFreeGatedWelcomeBack = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: true,
        qualities: false,
        services: false,
        subscriptionPlanViewed: true,
        freeGated: {
          hourlyRate: true,
          appDownload: true,
          welcomeBack: false,
        },
        firstName: null,
        lastName: null,
      },
    },
  },
};

const completenessSubscriptionPlanViewedComplete = {
  request: {
    query: GET_CAREGIVER_ENROLLMENT_STATUS,
    variables: {
      serviceType: 'CHILD_CARE',
    },
  },
  result: {
    data: {
      getCaregiverProfileCompleteness: {
        __typename: 'CaregiverProfileCompletenessStatus',
        additionalInfo: false,
        availability: true,
        bio: true,
        education: false,
        jobInterest: true,
        languages: true,
        overallStatus: 'INCOMPLETE',
        photo: true,
        qualities: false,
        services: false,
        subscriptionPlanViewed: true,
        freeGated: null,
        firstName: null,
        lastName: null,
      },
    },
  },
};

async function renderComponent(
  initialState = initialStateClone,
  mocks: MockedResponse<Record<string, any>>[] = [caregiverMock]
) {
  const result = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <AppStateProvider initialStateOverride={initialState}>
          <FeatureFlagsProvider flags={featureFlagsMockTrue}>
            <Preferences />
          </FeatureFlagsProvider>
        </AppStateProvider>
      </ThemeProvider>
    </MockedProvider>
  );
  await act(() => new Promise<undefined>((resolve) => setTimeout(resolve, 0)));
  return result;
}

describe('PreferencesPage', () => {
  let mockRouter: any = null;
  beforeEach(async () => {
    window.HTMLElement.prototype.scrollIntoView = () => {};
    mockRouter = {
      push: jest.fn(), // the component uses `router.push` only
      pathname: '/provider/cc',
    };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
  });

  afterEach(() => {
    // cleanup on exiting
    mockRouter = null;
  });

  it('matches snapshot', async () => {
    const renderSnapshotResult = await renderComponent();
    expect(renderSnapshotResult.asFragment()).toMatchSnapshot();
  });
  it('renders correctly', async () => {
    await renderComponent();
    expect(screen.getByText('Who would you like to care for?')).toBeInTheDocument();
  });
  it('should be able to select "Ages" all that apply', async () => {
    const { container } = await renderComponent();
    const pill = container.querySelector('input[value="EARLY_SCHOOL"]') as Element;
    await waitFor(() => expect(pill).not.toBeChecked());
    fireEvent.click(screen.getByText('4 - 5 yrs'));
    const btn = screen.getByRole('button', { name: 'Next' });
    fireEvent.click(btn);
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledTimes(1));
  });
  it('should be able to select number of children willling to care', async () => {
    const { container } = await renderComponent();
    const increment = container.querySelector('#increment') as Element;
    fireEvent.click(increment);
    expect(screen.getByText('Children')).toBeInTheDocument();
  });
  it('should able to redirect with out any selection', async () => {
    await renderComponent();
    const btn = screen.getByRole('button', { name: 'Next' });
    fireEvent.click(btn);
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledTimes(1));
  });
  it('should be able to select willing to care for sick children', async () => {
    await renderComponent();
    const sickChildrenCheckbox = screen.getByRole('checkbox', {
      name: 'Willing to care for sick children',
    });
    await waitFor(() => expect(sickChildrenCheckbox).not.toBeChecked());
    fireEvent.click(sickChildrenCheckbox);
  });
  it('should redirect to job-types page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessIncomplete]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.JOB_TYPES)
    );
  });
  it('should redirect to availability page when the caregiver profile is incomplete', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessJobInterestComplete]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.AVAILABILITY)
    );
  });
  it('should redirect to profile page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessAvailabilityComplete]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.PROFILE)
    );
  });
  it('should redirect to bio page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessProfileComplete]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.BIO)
    );
  });
  it('should redirect to photo page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessBioComplete]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.PHOTO)
    );
  });
  it('should redirect to preferences free gated page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessFreeGatedHourlyRate]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.PREFERENCES)
    );
  });
  it('should redirect to app download page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessFreeGatedAppDownload]);

    await waitFor(() =>
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.APP_DOWNLOAD)
    );
  });
  it('should redirect to welcome back page', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessFreeGatedWelcomeBack]);

    await waitFor(() =>
      // TODO: uncomment WELCOME_BACK route condition when BE is fixed
      // expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.WELCOME_BACK)
      expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_CHILD_CARE_ROUTES.APP_DOWNLOAD)
    );
  });
  it('should redirect to czen page when photo submitted', async () => {
    await renderComponent(initialStateClone, [caregiverMock, completenessPhotoComplete]);

    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(CZEN_BACKGROUND_CHECK_CC));
  });
  it('should redirect to czen page', async () => {
    await renderComponent(initialStateClone, [
      caregiverMock,
      completenessSubscriptionPlanViewedComplete,
    ]);

    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(CZEN_BACKGROUND_CHECK_CC));
  });
});
