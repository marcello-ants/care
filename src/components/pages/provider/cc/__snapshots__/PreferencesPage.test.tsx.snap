// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-17 MuiTypography-h2"
      >
        Who would you like to care for?
      </h2>
    </div>
    <div
      class="MuiGrid-root makeStyles-children-15 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="makeStyles-stepper-20 makeStyles-numberOfChildren-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-18 MuiInputLabel-animated"
            >
              Iâ€™m comfortable caring for:
            </label>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              class="makeStyles-stepperButtons-22 limit"
              disabled=""
              id="decrement"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-63"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  fill="#3b71aa"
                  r="11.5"
                />
                <path
                  d="M16.58 12.42H7.42a.42.42 0 010-.84h9.16a.42.42 0 110 .84z"
                  fill="#fff"
                  stroke="#fff"
                  stroke-miterlimit="10"
                  stroke-width=".7"
                />
              </svg>
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <input
              class="makeStyles-stepperInput-23"
              name="children"
              readonly=""
              type="text"
              value="1"
            />
          </div>
          <div
            class="MuiGrid-root makeStyles-addBlock-21 MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              class="makeStyles-stepperButtons-22"
              id="increment"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-root-64"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  fill="#3b71aa"
                  r="11.5"
                />
                <path
                  d="M12 17a.57.57 0 01-.57-.57V7.57a.57.57 0 011.14 0v8.86A.57.57 0 0112 17z"
                  fill="#fff"
                  stroke="#fff"
                  stroke-miterlimit="10"
                  stroke-width=".4"
                />
                <path
                  d="M16.43 12.57H7.57a.57.57 0 010-1.14h8.86a.57.57 0 010 1.14z"
                  fill="#fff"
                  stroke="#fff"
                  stroke-miterlimit="10"
                  stroke-width=".4"
                />
              </svg>
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-optionLabel-19 MuiInputLabel-animated"
            >
              Child
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-title-10 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Ages
      </h4>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root makeStyles-body3-29 MuiTypography-body1 MuiTypography-colorSecondary"
      >
        Select all that apply
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-rootInline-40 makeStyles-selector-13 makeStyles-allItemsSelector-14 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-41 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-44"
            id="selector.undefined.0"
            name=""
            readonly=""
            type="checkbox"
            value="NEWBORN"
          />
          <div
            class="MuiBox-root MuiBox-root-52 makeStyles-controlIndicator-45 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              0 - 11 mos
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-41 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-44"
            id="selector.undefined.1"
            name=""
            readonly=""
            type="checkbox"
            value="TODDLER"
          />
          <div
            class="MuiBox-root MuiBox-root-53 makeStyles-controlIndicator-45 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              1 - 3 yrs
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-41 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-44"
            id="selector.undefined.2"
            name=""
            readonly=""
            type="checkbox"
            value="EARLY_SCHOOL"
          />
          <div
            class="MuiBox-root MuiBox-root-54 makeStyles-controlIndicator-45 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              4 - 5 yrs
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-41 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-44"
            id="selector.undefined.3"
            name=""
            readonly=""
            type="checkbox"
            value="ELEMENTARY_SCHOOL"
          />
          <div
            class="MuiBox-root MuiBox-root-55 makeStyles-controlIndicator-45 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              6 - 11 yrs
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-41 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-44"
            id="selector.undefined.4"
            name=""
            readonly=""
            type="checkbox"
            value="TEEN"
          />
          <div
            class="MuiBox-root MuiBox-root-56 makeStyles-controlIndicator-45 sm"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              12+ yrs
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        class="MuiFormControlLabel-root makeStyles-checkboxLabel-11"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-58 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-61"
              data-indeterminate="false"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-65"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                fill="#fff"
              />
              <path
                d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                fill="#23405f"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Willing to care for sick children
        </span>
      </label>
    </div>
    <div
      class="MuiGrid-root makeStyles-nextButtonContainer-2 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
